File: skellig-test-step-processing/src/test/java/org/skellig/teststep/processing/validation/comparator/RegexValueComparatorTest.java
package org.skellig.teststep.processing.validation.comparator;

import org.junit.jupiter.api.BeforeEach;
import org.junit.jupiter.api.Test;

import static org.junit.jupiter.api.Assertions.assertFalse;
import static org.junit.jupiter.api.Assertions.assertTrue;

class RegexValueComparatorTest {

    private RegexValueComparator regexValueComparator;

    @BeforeEach
    void setUp() {
        regexValueComparator = new RegexValueComparator();
    }

    @Test
    void testRegexMatches() {
        assertTrue(regexValueComparator.compare("regex(\\w+\\s?=\\s?\\w+)", "a1 = v1"));
    }

    @Test
    void testRegexNotMatches() {
        assertFalse(regexValueComparator.compare("regex(\\d+\\s?=\\s?\\d+)", "a1 = v1"));
    }

    @Test
    void testRegexWithNullActualResult() {
        assertFalse(regexValueComparator.compare("regex(.*)", null));
    }

    @Test
    void testInvalidRegex() {
        assertFalse(regexValueComparator.compare("regex((??)", "value"));
    }
}

File: skellig-test-step-processing/src/test/kotlin/org/skellig/teststep/processing/validation/comparator/RegexValueComparatorTest.kt
package org.skellig.teststep.processing.validation.comparator

import org.junit.jupiter.api.Assertions
import org.junit.jupiter.api.BeforeEach
import org.junit.jupiter.api.Test

class RegexValueComparatorTest {

    private var regexValueComparator: RegexValueComparator? = null

    @BeforeEach
    fun setUp() {
        regexValueComparator = RegexValueComparator()
    }

    @Test
    fun testRegexMatches() {
        Assertions.assertTrue(regexValueComparator!!.compare("regex(\\w+\\s?=\\s?\\w+)", "a1 = v1"))
    }

    @Test
    fun testRegexNotMatches() {
        Assertions.assertFalse(regexValueComparator!!.compare("regex(\\d+\\s?=\\s?\\d+)", "a1 = v1"))
    }

    @Test
    fun testRegexWithNullActualResult() {
        Assertions.assertFalse(regexValueComparator!!.compare("regex(.*)", null))
    }

    @Test
    fun testInvalidRegex() {
        Assertions.assertFalse(regexValueComparator!!.compare("regex((??)", "value"))
    }
}